src/test/rpc/Peers_test.cpp

```diff
@@ -0,0 +1,98 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/Env.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/overlay/Cluster.h>
+#include <ripple/overlay/Overlay.h>
+#include <unordered_map>
+
+namespace ripple {
+
+class Peers_test : public beast::unit_test::suite
+{
+    void testRequest()
+    {
+        testcase("Basic request");
+        using namespace test::jtx;
+        Env env {*this};
+
+        // without modification of the cluster, expect an empty set
+        // from this request
+        auto peers = env.rpc("peers")[jss::result];
+        BEAST_EXPECT(peers.isMember(jss::cluster) &&
+            peers[jss::cluster].size() == 0 );
+        BEAST_EXPECT(peers.isMember(jss::peers) &&
+            peers[jss::peers].isNull());
+
+        // insert some nodes in to the cluster
+        std::unordered_map<std::string, std::string> nodes;
+        for(auto i =0; i < 3; ++i)
+        {
+
+            auto seed = generateSeed("seed" + std::to_string(i));
```

Comment:
Not change needed, but you could replace 50-53 with:

    auto kp = generateKeyPair (KeyType::secp256k1, generateSeed("seed" + std::to_string(i)));

You'd also have the `SecretKey` in `kp` which you don't strictly need, but, overall, I think it would simpify the code.